### PR TITLE
Document Pack's Component

### PR DESCRIPTION
**Pull request by karl-cardenas-coding**
## Describe the Change

This PR documents behaviors that operate to the new Pack component and how to handle edge cases. [Link to README preview](https://github.com/spectrocloud/librarium/tree/packs-docs#packs-component).

## Changed Pages

💻 [Preview URL for Page](https://github.com/spectrocloud/librarium/tree/packs-docs#packs-component)

## Jira Tickets

🎫 [DOC-1283](https://spectrocloud.atlassian.net/browse/DOC-1283)

## Backports

Can this PR be backported?

- [ ] Yes. _Remember to add the relevant backport labels to your PR._
- [x] No. _Please leave a short comment below about why this PR cannot be backported._

[DOC-1283]: https://spectrocloud.atlassian.net/browse/DOC-1283?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -719,6 +720,109 @@ This is a <VersionedLink text="Internal Link" url="/getting-started/additional-c
...
 `..`. External links can be referenced as usual.
 
+## Packs Component
+
+The packs component is a custom component that displays all packs available in Palette SaaS by querying the Palette API
+at `api.spectrocloud.com`. The component is powered by a [custom plugin](./plugins/packs-integrations.js) that fetches
+the data before a build or a local development server start.
+
+> [!NOTE]
+>
```
Comment (caroldelwing) [inline README.md:730]:
```suggestion
```
For the callout box to work :)

File: README.md [hunk 2]
```diff
@@ -719,6 +720,109 @@ This is a <VersionedLink text="Internal Link" url="/getting-started/additional-c
...
+> The data is stored in the file `.docusarus/packs-integration/api_pack_response.json` once all the packs are fetched.
+> The logos are stored in `static/img/packs/`. If you remove the `.docusarus` folder, the plugin will fetch the packs
+> again. If the pack's data is present, the plugin will skip re-downloading the data. If you want to remove the packs
+> data and trigger a new fetch, you > can issue the following command `make clean-packs`.
```
Comment (caroldelwing) [inline README.md:734]:
```suggestion
> data and trigger a new fetch, you can issue the command `make clean-packs`.
```

File: README.md [hunk 3]
```diff
@@ -719,6 +720,109 @@ This is a <VersionedLink text="Internal Link" url="/getting-started/additional-c
...
+
+Pack descriptions are stored in JSON file maintained by the documentation team. The JSON file is located at
```
Comment (caroldelwing) [inline README.md:736]:
```suggestion
Pack descriptions are stored in a JSON file maintained by the documentation team. The JSON file is located at
```

File: README.md [hunk 4]
```diff
@@ -719,6 +720,109 @@ This is a <VersionedLink text="Internal Link" url="/getting-started/additional-c
...
+
+```json
+{
+  "name": "insert the name of the pack. DO NOT USE THE DISPLAY NAME",
+  "description": "insert description here"
+}
+```
+
+Some things to keep in mind related to descriptions. If the local development server is active and you make changes to
+the `/static/packs-data/packs_information.json`file. The changes will not be reflected in the pack component. You must
```
Comment (caroldelwing) [inline README.md:748]:
```suggestion
the `/static/packs-data/packs_information.json`file, the changes will not be reflected in the pack component. You must
```

File: README.md [hunk 5]
```diff
@@ -719,6 +720,109 @@ This is a <VersionedLink text="Internal Link" url="/getting-started/additional-c
...
+stop and restart the local development server to observe the changes. The same applies to the build process. The other
+item to remember is to reference a pack by its name, not the display name. The name is the pack's name, and it is
+referenced in the Palette API. You can find the name in the description component or by looking at the URL of the pack's
+page.
```
Comment (caroldelwing) [inline README.md:752]:
```suggestion
stop and restart the local development server to observe the changes. The same applies to the build process. Another thing to remember is to reference a pack by the name used in the Palette API, not the display name. You can find the pack's name in the description component or by looking at the URL of the pack's page.
```
Maybe? :)

File: README.md [hunk 6]
```diff
@@ -719,6 +720,109 @@ This is a <VersionedLink text="Internal Link" url="/getting-started/additional-c
...
+
+#### README Content
+
+The pack component will display a Pack's README file if it exists. The README content comes from the Palette API.
+Depending on the available content, the packs component will display the README content, the additional details content,
+or a message indicating no available content. Refer to the table below for the different scenarios.
```
Comment (caroldelwing) [inline README.md:758]:
```suggestion
or a message indicating that no content is available. Refer to the table below for the different scenarios.
```

File: README.md [hunk 7]
```diff
@@ -719,6 +720,109 @@ This is a <VersionedLink text="Internal Link" url="/getting-started/additional-c
...
+| ❌               | ❌                                | A message indicating that no content is available.       |
+
+#### Additional Details Content
+
+To display the Additional Details content, create a markdown file with the same name as the pack in the the
+[`docs/docs-content/integrations/`](./docs/docs-content/integrations/) content folder. For example, if the pack name is
+`ubuntu-aws`, you would create a markdown file called `ubuntu-aws.md`. The additional details content requires you to
+follow the [Packs layout guide](https://spectrocloud.atlassian.net/wiki/spaces/DE/pages/1802797059/Packs).
+
+If you want to add content specific to a version, include the heading and tabs component in the markdown file
```
Comment (caroldelwing) [inline README.md:774]:
```suggestion
If you want to add content specific to a version, include the following heading and tabs component in the markdown file.
```

File: README.md [hunk 8]
```diff
@@ -719,6 +720,109 @@ This is a <VersionedLink text="Internal Link" url="/getting-started/additional-c
...
+
+<Tabs queryString="parent"> -> Tabs for different versions
+<TabItem label="1.1.x" value="1.1.2">
+ Insert content here as needed. Create more `TabItem` components as needed.
+</TabItem>
+</Tabs>
+```
+
+> [!WARNING]
+>
```
Comment (caroldelwing) [inline README.md:788]:
```suggestion
```

File: README.md [hunk 9]
```diff
@@ -719,6 +720,109 @@ This is a <VersionedLink text="Internal Link" url="/getting-started/additional-c
...
+> not be displayed correctly, and as a result, actual tabs will be displayed, and the pack will lose the ability to
+> automatically display the content for the select version the user has specified in the version drop-down.
+
+The packs component will always display the top-level tab content, so if you add content for a new version, ensure the
+new `TabItem` component is the first in the list. If a new pack version does not have a respective tag, the latest
+version content will be displayed automatically.
+
+#### Links
+
+When authoring additional details content, you must use the `<VersionedLink />` component to link to other documentation
```
Comment (caroldelwing) [inline README.md:800]: Do you mean partials or any content?

File: README.md [hunk 10]
```diff
@@ -719,6 +720,109 @@ This is a <VersionedLink text="Internal Link" url="/getting-started/additional-c
...
+pages. The component is required to ensure that the links are versioned correctly. Refer to the
+[Internal Links](#internal-links) section for more information.
+
+If you want to link to a heading inside the pack component, must also use the `<VersionedLink />` and include the path
```
Comment (caroldelwing) [inline README.md:804]:
```suggestion
If you want to link to a heading inside the pack component, you must also use the `<VersionedLink />` and include the path
```

**Review by caroldelwing — APPROVED** [review]
Super complete! I left small suggestions :)

File: README.md (hunk 9 above)
Comment (karl-cardenas-coding) [reply README.md:800]: Nope

**Pull request merged**